src/overlays/actors/ovl_Bg_Hakugin_Elvpole/z_bg_hakugin_elvpole.c

```diff
@@ -30,17 +30,111 @@ const ActorInit Bg_Hakugin_Elvpole_InitVars = {
+    this->unk_15C = 0;
+    this->unk_160 = false;
+    if ((switchFlags != 0x7F) && (Flags_GetSwitch(play, switchFlags))) {
+        this->unk_15E = 0x64;
+        return;
+    }
+    this->unk_15E = -1;
+    this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 320.0f;
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Init.s")
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Destroy.s")
+void BgHakuginElvpole_Destroy(Actor* thisx, PlayState* play) {
+    BgHakuginElvpole* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Destroy.s")
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/func_80ABD92C.s")
+// #pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/func_80ABD92C.s")
+void func_80ABD92C(BgHakuginElvpole* this, PlayState* play) {
+    s32 pad;
+    s32 sp28;
+    f32 var_fv1;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_Bg_Hakugin_Elvpole/BgHakuginElvpole_Update.s")
+    sp28 = false;
+    if (this->unk_15E > 0) {
+        if ((this->dyna.actor.world.pos.y - this->dyna.actor.home.pos.y) < (-120.0f)) {
+            this->dyna.actor.world.pos.y += 4.0f;
+            sp28 = true;
+        } else if (this->unk_15E > 0) {
+            if (this->unk_15E == 100) {
+                Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_STONEDOOR_STOP);
+            }
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 120.0f;
+            // FAKE
+            if (1) {}
+            this->unk_15E--;
+        }
+    } else if (this->unk_15E == 0) {
+        if ((this->dyna.actor.world.pos.y - this->dyna.actor.home.pos.y) > -320.0f) {
+            this->dyna.actor.world.pos.y -= 4.0f;
+            sp28 = true;
+        } else {
+            this->dyna.actor.world.pos.y = this->dyna.actor.home.pos.y - 320.0f;
+            Flags_UnsetSwitch(play, BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor));
+            this->unk_15E = -1;
+            Actor_PlaySfxAtPos(&this->dyna.actor, NA_SE_EV_STONEDOOR_STOP);
+        }
+    } else {
+        if (((BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor)) != 0x7F) &&
+            (Flags_GetSwitch(play, BGHAKUGINELVPOLE_GET_SWITCHFLAG(&this->dyna.actor)))) {
+            this->unk_15E = 0x64;
+            this->unk_160 = true;
+        }
+    }
+    if ((sp28) || ((this->unk_15C & 7) != 0)) {
+        if (this->unk_15C & 1) {
+            var_fv1 = 1.0f;
+        } else {
+            var_fv1 = -1.0f;
+        }
+        this->unk_15C++;
+        this->dyna.actor.world.pos.x = (Math_SinS(this->unk_15C * 0x2000) * var_fv1) + this->dyna.actor.home.pos.x;
+        this->dyna.actor.world.pos.z = (Math_CosS(this->unk_15C * 0x2000) * var_fv1) + this->dyna.actor.home.pos.z;
+        func_800B9010(&this->dyna.actor, 0x2103U);
```

Comment:
```suggestion
        func_800B9010(&this->dyna.actor, NA_SE_EV_HORSE_WALK - SFX_FLAG);
```

Reply:
It's NA_SE_EV_PLATE_LIFT_LEVEL - SFX_FLAG, not HORSE_WALK, but regardless, fixed